multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQCompactionRunner.java

```diff
@@ -293,6 +305,22 @@ public List<MSQControllerTask> createMsqControllerTasks(
     return msqControllerTasks;
   }
 
+  private Query<?> maybeApplyPolicy(DataSchema dataSchema, Query<?> query)
```

Comment:
Nit: Please add a short javadoc too.

```suggestion
  private Query<?> maybeApplyPolicyOnDatasource(DataSchema dataSchema, Query<?> query)
```

Reply:
i didn't think this was needed since `Policy` apply to datasources so it seemed redundant, but can rename if you feel like it isn't obvious

Reply:
this is changed now to just be `getInputDataSource` and is used by the 'build' query methods directly to create the `DataSource` instead of rewriting the query later